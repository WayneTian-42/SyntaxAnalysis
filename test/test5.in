S -> (T) | aA
A -> +S | #
T -> SB
B -> ,SB | #