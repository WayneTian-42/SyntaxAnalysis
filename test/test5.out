First Set
A  # + 
B  # , 
S  ( a 
T  ( a 

Follow Set
A  $ ) , 
B  ) 
S  $ ) , 
T  ) 

Predict Analysis Table
        (              )              +              ,              a              $              
S       S -> (T)                                                    S -> aA                       
A                      A -> #         A -> +S        A -> #                        A -> #         
T       T -> SB                                                     T -> SB                       
B                      B -> #                        B -> ,SB                                     

Please enter the symbols:
a+(a,a))

Stack               Input               Output              
$S                  a+(a,a))$           S -> aA             
$Aa                 a+(a,a))$           accept a
$A                  +(a,a))$            A -> +S             
$S+                 +(a,a))$            accept +
$S                  (a,a))$             S -> (T)            
$)T(                (a,a))$             accept (
$)T                 a,a))$              T -> SB             
$)BS                a,a))$              S -> aA             
$)BAa               a,a))$              accept a
$)BA                ,a))$               A -> #              
$)B                 ,a))$               B -> ,SB            
$)BS,               ,a))$               accept ,
$)BS                a))$                S -> aA             
$)BAa               a))$                accept a
$)BA                ))$                 A -> #              
$)B                 ))$                 B -> #              
$)                  ))$                 accept )
$                   )$                  error $ )